src/api/Startup.cs

```diff
@@ -105,20 +108,24 @@ public class Startup {
         }
 
         public void ConfigureContainer(ContainerBuilder builder) {
-            builder.RegisterAssemblyTypes(typeof(IMediator).GetTypeInfo().Assembly).AsImplementedInterfaces();
+            // set up computations
+            builder.RegisterAssemblyTypes(typeof(Startup).GetTypeInfo().Assembly)
+                    .AsClosedTypesOf(typeof(IComputationHandler<,>))
+                    .AsImplementedInterfaces();
 
-            var mediatrOpenTypes = new[]
-            {
-                typeof(IRequestHandler<,>),
-                typeof(INotificationHandler<>),
-            };
+            builder.RegisterDecorator<DoubleAvenuesException.Decorator, IComputationHandler<ZoneParsing.Computation, AddressWithGrids>>();
```

Comment:
**thought**: i'm not super happy about registering these in here. It would be more normal to do it in ConfigureServices but `RegisterDecorator` is only available on the autofac builder